src/Agent.Sdk/Util/StringUtil.cs

```
@@ -291,6 +292,43 @@ public static string DeactivateVsoCommands(string input)
                 return string.Empty;
             }
 
+            input = DeactivateBase64EncodedVsoCommands(input);
+            return ScrapVsoCommands(input);
+        }
+
+        /// <summary>
+        /// Tries to decode the input string assuming it to be base64 encoded and
+        /// scraps vso command if any and re-encodes the updated string.
+        /// An exception is thrown for the case when the input is not base64 encoded and input is returned unmodified.
+        /// </summary>
+        /// <returns>String without vso commands that can be executed</returns>
+        public static string DeactivateBase64EncodedVsoCommands(string input)
+        {
+            if (input == null)
+            {
+                throw new ArgumentNullException(nameof(input), "Input string cannot be null.");
+            }
+
+            if (input.Length == 0)
+            {
+                return string.Empty;
+            }
+
+            byte[] buffer = new byte[input.Length];
+            if (Convert.TryFromBase64String(input, buffer, out int bytesWritten))
```

TryFromBase64String should be used when we are not always certain if the conversion would work. However, you are authoring a utility method called `DeactivateBase64EncodedVsoCommands`. The caller would have no clue if the conversion failed as you are returning the input as is when the conversion fails. Furthermore, failure of this conversion is not even logged making debugging even more difficult.

It is suggested to attempt conversion normally and throw the exception from this method. Let the caller decide what they want to do with it. Either they handle it or terminate the flow depending on their scenario.

---

Updated the function name

---

Glad that the exception is being thrown instead of returning the input as is when the conversion fails. And here is a simplified version of the function. 

```
  public static string DeactivateVsoCommandsIfBase64Encoded(string input)
      {
          if (input == null)
          {
              throw new ArgumentNullException(nameof(input), "Input string cannot be null.");
          }
          if (input.Length == 0)
          {
              return string.Empty;
          }
          byte[] decodedBytes = Convert.FromBase64String(input);
          string decodedString = Encoding.UTF8.GetString(decodedBytes);
          decodedString = ScrapVsoCommands(decodedString);
          return Convert.ToBase64String(Encoding.UTF8.GetBytes(decodedString));
      }
```